doct.el

```diff
@@ -1282,6 +1283,87 @@ Otherwise, add to front."
   (let ((args (list (doct declarations) (copy-tree list))))
     (apply #'append (if append (nreverse args) args))))
 
+
+;; A structure for captures
+;; PARENTS is a list of strings which are the names of the capture's parents
+;; CHILD is t if the capture is a child of another and nil if it isn't
+;; DECLARATION is the declaration of the capture in doct
+(cl-defstruct doct--capture
+  parents
+  child
+  declaration)
+
+;; The capture namespace. Hash-keys are strings.
+(defvar doct--capture-namespace (make-hash-table :test 'equal))
```

Comment:
Org's capture templates are unique only in the keys used to select them. Because of this, multiple declarations may have the same name. Indexing the hash by declaration name would not allow this. We would also need to know the full sequence of inherited keys, too, as a child may have the same declared keys and name as a child that is part of another hierarchy. An example illustrating both problems:

```emacs-lisp
(doct '( :group "Example"
         :file ""
         :children
         (("Parent One" :keys "1"
           ;; same name and keys as Parent Two's "Child"
           :children ("Child" :keys "c"))
          ("Parent Two" :keys "2"
           ;; same name and keys as Parent One's "Child"
           :children ("Child" :keys "c")))))
```

Unless I'm mistaken, the current implementation would have a namespace collision here:

```emacs-lisp
(doct-defcapture "Child" ("Parent One") :keys "c" :file "")
(doct-defcapture "Child" ("Parent Two") :keys "c" :file "")
```

This could probably be worked around by taking the declaration's parents into consideration
when hashing it.

Reply:
A simple solution is to index declarations in the hash table by a user-defined symbol just like `defun' does it:

```emacs-lisp
(doct-declare identifier (&rest parents)
  "Name string or (short) description"
  :keys "c" ...)
```

Where `identifier` is now a symbol instead of a string corresponding to a declaration's name. The rest of the machinery would work with a few little modifications.